bundles/azure/aso/psql/bundle/config/01-flexible-server.ytt.yml

```diff
@@ -0,0 +1,229 @@
+#@ end
+---
+apiVersion: secretgen.k14s.io/v1alpha1
+kind: Password
+metadata:
+  name: #@ dbSecretName
+  namespace: #@ data.values.resource_namespace
+spec:
+  length: 64
+  secretTemplate:
+    type: Opaque
+    stringData:
+      password: $(value)
+---
+apiVersion: resources.azure.com/v1beta20200601
+kind: ResourceGroup
+metadata:
+  name: #@ resourceGroupK8SName
+  namespace: #@ data.values.resource_namespace
+  annotations:
+    serviceoperator.azure.com/operator-namespace: #@ data.values.aso_controller_namespace
+#@ if data.values.use_existing_resource_group:
+    serviceoperator.azure.com/reconcile-policy: skip
+#@ end
+  labels:
+    kapp.k14s.io/noop: ""
+spec:
+  azureName: #@ data.values.resource_group
+  location: #@ data.values.location
+
+#@ databaseCounter = 0
+#@ for db in data.values.databases:
+#@   dbName = db.name
+#@   if dbName == "":
+#@     dbName = flexibleServerName + "-" + str(databaseCounter)
+#@     databaseCounter += 1
+#@   end
+---
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServersDatabase
+metadata:
+  name: #@ dbName
+  namespace: #@ data.values.resource_namespace
+spec:
+  azureName: #@ dbName
+  owner:
+    name: #@ flexibleServerName
+  charset: #@ db.charset
+  collation: #@ db.collation
+#@ end
+
+#@ fwRuleCounter = 0
+#@ for val in data.values.firewall_rules:
+#@   startAddr = ip.parse_addr(val.startIp)
+#@   endAddr = ip.parse_addr(val.endIp)
+--- 
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServersFirewallRule
+metadata:
+  name: #@ data.values.resource_name + "-" + str(fwRuleCounter)
+  namespace: #@ data.values.resource_namespace
+spec:
+  owner:
+    name: #@ flexibleServerName
+  startIpAddress: #@ startAddr.string()
+  endIpAddress: #@ endAddr.string()
+#@   fwRuleCounter += 1
+#@ end  
+---
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServer
+metadata:
+  name: #@ flexibleServerName
+  namespace: #@ data.values.resource_namespace
+spec:
+  location: #@ data.values.location
+  azureName: #@ flexibleServerName
+  owner:
+    name: #@ resourceGroupK8SName
+  version: #@ data.values.version
+  sku:
+    name: #@ data.values.instance_type
+    tier: #@ data.values.instance_tier
+  administratorLogin: #@ data.values.administrator_name
+  administratorLoginPassword:
+    name: #@ dbSecretName
+    key: password
+  storage:
+    storageSizeGB: #@ data.values.instance_storage_size_gb
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: #@ serviceAccountName
+  namespace: #@ serviceAccountNamespace
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: #@ roleName
+  namespace: #@ data.values.resource_namespace
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+- apiGroups:
+  - dbforpostgresql.azure.com
+  resources:
+  - flexibleservers
+  - flexibleserversdatabases
+  - flexibleserversfirewallrule
```

Comment:
*nitpick* I don't think SecretTemplate needs access to flexibleserversfirewallrule resources.

Reply:
Agreed.

Reply:
We cleaned it up.